### PR TITLE
HOLO-1418: Estimate gas for collection deploys, and add more tests to collection classes

### PR DESCRIPTION
**Pull request by feliperodriguess**
## Describe Changes

Changes made to the package: **sdk**.

I made it better by doing ...

- Create `_estimateGasForDeployingCollection` method for `HolographCollectionLegacy` class
- Add tests to `collection-legacy.ts` file
- Create `_estimateGasForDeployingCollection` method for `HolographMoeERC721DropV1` and `V2` classes
- Add tests to `collection-moe.ts` file
- Add docs
- General improvements

image

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All tests are passing

### CODE REVIEW

File: packages/sdk/src/assets/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#### TODO: Move this content to our SDK docs website soon
```
Comment (sogoiii) [inline packages/sdk/src/assets/README.md:1]: This is a great start! I do like the ordering. We will translate this to the doc sites.

File: packages/sdk/src/assets/collection-legacy.ts [hunk 2]
```diff
@@ -96,29 +99,31 @@ export class HolographLegacyCollection {
     return this.collectionInfo
   }
 
-  // TODO: Make all _* methods private after setting up all tests
-  async _getFactoryAddress() {
-    const factoryAddress = await this.factory.getAddress(this.primaryChainId)
+  async _getFactoryAddress(chainId = this.primaryChainId) {
+    const factoryAddress = await this.factory.getAddress(chainId)
```
[the remark this answers is not in the document]
Comment (feliperodriguess) [reply packages/sdk/src/assets/collection-legacy.ts:103]: Done ✅

File: packages/sdk/src/assets/collection.validation.ts [hunk 3]
```diff
@@ -117,3 +117,117 @@ export const validate = {
   nftIpfsUrl: nftIpfsUrlSchema,
   nftIpfsImageCid: nftIpfsImageCidSchema,
 }
+
+export const DROP_INIT_CODE_ABI_PARAMETERS = {
+  [HolographVersion.V1]: [
+    {
```
[the remark this answers is not in the document]
Comment (feliperodriguess) [reply packages/sdk/src/assets/collection.validation.ts:123]: Done ✅

File: packages/sdk/src/constants/addresses.ts [hunk 4]
```diff
@@ -126,7 +126,7 @@ export class Addresses {
       case Environment.localhost: {
         switch (chainId) {
           default:
-            return ''.toLowerCase() as Address // TODO: Add address
+            return '0xdF26982B2D5A4904757f6099b939c0eBcFE70668'.toLowerCase() as Address
```
[the remark this answers is not in the document]
Comment (feliperodriguess) [reply packages/sdk/src/constants/addresses.ts:129]: EditionsMetadataRendererProxy took from localhost deployments, just followed the pattern as the others in this file

**Review by sogoiii — APPROVED** [review]
👍

**Review by BravoNatalie — APPROVED** [review]

**Pull request merged**